sormas-app/app/src/main/java/de/symeda/sormas/app/backend/vaccination/VaccinationEntity.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2020 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package de.symeda.sormas.app.backend.vaccination;
+
+import static de.symeda.sormas.api.EntityDto.COLUMN_LENGTH_BIG;
+
+import java.util.Date;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+
+import com.j256.ormlite.field.DataType;
+import com.j256.ormlite.field.DatabaseField;
+import com.j256.ormlite.table.DatabaseTable;
+
+import de.symeda.sormas.api.caze.Trimester;
+import de.symeda.sormas.api.caze.VaccinationInfoSource;
+import de.symeda.sormas.api.caze.Vaccine;
+import de.symeda.sormas.api.caze.VaccineManufacturer;
+import de.symeda.sormas.api.utils.YesNoUnknown;
+import de.symeda.sormas.app.backend.clinicalcourse.HealthConditions;
+import de.symeda.sormas.app.backend.common.EmbeddedAdo;
+import de.symeda.sormas.app.backend.common.PseudonymizableAdo;
+import de.symeda.sormas.app.backend.immunization.Immunization;
+import de.symeda.sormas.app.backend.user.User;
+
+@Entity(name = VaccinationEntity.TABLE_NAME)
+@DatabaseTable(tableName = VaccinationEntity.TABLE_NAME)
+@EmbeddedAdo
+public class VaccinationEntity extends PseudonymizableAdo {
+
+	public static final String TABLE_NAME = "vaccination";
+	public static final String I18N_PREFIX = "vaccination";
+
+	@DatabaseField(foreign = true, foreignAutoRefresh = true, canBeNull = false)
+	private Immunization immunization;
+	@DatabaseField(foreign = true, foreignAutoRefresh = true, canBeNull = false)
+	private HealthConditions healthConditions;
+	@DatabaseField(dataType = DataType.DATE_LONG, canBeNull = false)
+	private Date reportDate;
+	@DatabaseField(foreign = true, foreignAutoRefresh = true)
+	private User reportingUser;
+	@DatabaseField(dataType = DataType.DATE_LONG)
+	private Date vaccinationDate;
+	@Enumerated(EnumType.STRING)
+	private Vaccine vaccineName;
+	@Column(length = COLUMN_LENGTH_BIG)
```

Comment:
Use `@Column(columnDefinition = "text")` for all fields that have the data type "text" in the database (so everything that you've declared as COLUMN_LENGTH_BIG here)